libs/model/src/user/UpdateUser.command.ts

```diff
@@ -0,0 +1,116 @@
+import { InvalidInput, type Command } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { DEFAULT_NAME } from '@hicommonwealth/shared';
+import { models } from '../database';
+import { mustExist } from '../middleware/guards';
+import { getDelta, sanitizeQuillText, updateTags } from '../utils';
+
+export function UpdateUser(): Command<typeof schemas.UpdateUser> {
+  return {
+    ...schemas.UpdateUser,
+    auth: [],
+    body: async ({ actor, payload }) => {
+      // comparing number to string since command convention requires string id
+      if (actor.user.id != payload.id)
+        throw new InvalidInput('Invalid user id');
+
+      const { id, profile, tag_ids } = payload;
+      const {
+        slug,
+        name,
+        email,
+        website,
+        avatar_url,
+        socials,
+        bio: rawBio,
+        background_image,
+      } = profile;
+
+      const user = (
+        await models.User.findOne({
+          where: { id },
+          include: {
+            model: models.ProfileTags,
+          },
+        })
+      )?.toJSON();
+      if (!mustExist('User', user)) return;
+
+      const is_welcome_onboard_flow_complete =
+        user.is_welcome_onboard_flow_complete ||
+        !!name ||
+        user.profile.name !== DEFAULT_NAME
+          ? true
+          : false;
```

Comment:
This is regressing, we should only set `is_welcome_onboard_flow_complete: true` when username existing, and that name is not `DEFAULT_NAME`, something like we did [previously](https://github.com/hicommonwealth/commonwealth/pull/8574/files#diff-68d374c2f8d0a0b1010d8573a18ebb7093d871b45b7d374818b9a770948a76d5L99-L108) 

```
name &&
name !== DEFAULT_NAME &&
!req.user.is_welcome_onboard_flow_complete
```

and lets also block `is_welcome_onboard_flow_complete` in request body, as we don't allow users to update this from frontend.